### ISSUE TITLE
Create balanced crawler behaviour

### ISSUE THREAD

**Issue by xdvom03** [xdvom03/klaus#58]
Drunkbot tried wandering completely randomly, considering all links. But many sites extensively link to other things within the same domain. If the bot cannot immediately find something relevant (which is most of the time for specific queries), it tends to get stuck in a local maximum (often of probability 10<sup>-3</sup> or less).

Wanderbot only considers domain jumps - only domains it has never been to. This works, though only after a lot of bungling around - it eventually finds something related, then keeps to useful links between domains. The problem with it is that once it acquires a target, it cannot continue to search it. Links to other domains presumably don't exist for all pages within the target domain, so lots of articles are inaccessible.

Clearly, a balance is needed. Giving sites on the same domain a multiplicative penalty does not fix the local maximum problem (at times, it can be a difference of many orders of magnitude, but that is irrelevant, below 10<sup>-3</sup> the chances might as well be zero). It also caps their score below 1, giving us the worst of both worlds.

A more hopeful alternative is to only consider same-domain links if they are x times (starting at 2, let's see what it does) closer to a probability of 1. They will only be ever considered if their score is over 0.5, negating the local maximum problem.

**Comment by xdvom03** [comment]
Superseded by #63, may be reconsidered once a new model is reached.

**Comment by xdvom03** [comment]
Nope, these issues are orthogonal. I think I'll allow Classbot to make a limited number of jumps within a single domain (thinking five to ten), alowing for some deeper content than homepages.

**Comment by xdvom03** [comment]
The inter-intra-domain system is sufficient - deep and wide web coverage. It does cause some links to die unnecessarily, and might thus need tweaking.
